### PR TITLE
fby3: vf: Initialize VernalFall BIC

### PR DESCRIPTION
**Pull request by HungYi-Li**
Summary:
- Initialize VernerFall BIC.
- Add project code in meta-facebook/yv3-vf/.
- define GPIO name and IPMB table

Test plan:
- Build code: PASS
- Can access BIC console: PASS

### CODE REVIEW

File: meta-facebook/yv3-vf/prj.conf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+CONFIG_NUM_PREEMPT_PRIORITIES=56
+CONFIG_HEAP_MEM_POOL_SIZE=32768
+CONFIG_CMSIS_V2_MSGQ_MAX_DYNAMIC_SIZE=8192
+CONFIG_POLL=y
+CONFIG_THREAD_NAME=y
+CONFIG_THREAD_STACK_INFO=y
+CONFIG_THREAD_MONITOR=y
+CONFIG_IRQ_OFFLOAD=y
+CONFIG_INIT_STACKS=y
+CONFIG_MAX_THREAD_BYTES=4
```
Comment (GoldenBug) [inline meta-facebook/yv3-vf/prj.conf:12]: This requires userspace. so it can be removed.

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: meta-facebook/yv3-vf/prj.conf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+CONFIG_POSIX_CLOCK=y
+CONFIG_STACK_SENTINEL=y
+CONFIG_LOG=y
+CONFIG_LOG_BACKEND_UART=y
+CONFIG_SHELL=y
+CONFIG_SHELL_BACKEND_SERIAL_INIT_PRIORITY=99
+CONFIG_IPMI=y
+CONFIG_WATCHDOG=y
+
+CONFIG_USERSPACE=y
```
[the remark this answers is not in the document]
Comment (HungYi-Li) [reply meta-facebook/yv3-vf/prj.conf]: disable it

File: meta-facebook/yv3-vf/src/platform/plat_gpio.h [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	gpio_name_to_num(Reserve_GPIOT0) gpio_name_to_num(Reserve_GPIOT1)                          \
+		gpio_name_to_num(Reserve_GPIOT2) gpio_name_to_num(Reserve_GPIOT3)                  \
+			gpio_name_to_num(Reserve_GPIOT4) gpio_name_to_num(Reserve_GPIOT5)          \
+				gpio_name_to_num(Reserve_GPIOT6) gpio_name_to_num(Reserve_GPIOT7)
+// GPIOU input only
+#define name_gpioU                                                                                 \
+	gpio_name_to_num(Reserve_GPIOU0) gpio_name_to_num(Reserve_GPIOU1)                          \
+		gpio_name_to_num(Reserve_GPIOU2) gpio_name_to_num(Reserve_GPIOU3)                  \
+			gpio_name_to_num(Reserve_GPIOU4) gpio_name_to_num(Reserve_GPIOU5)          \
+				gpio_name_to_num(Reserve_GPIOU6) gpio_name_to_num(Reserve_GPIOU7)
```
[the remark this answers is not in the document]
Comment (HungYi-Li) [reply meta-facebook/yv3-vf/src/platform/plat_gpio.h]: Add clang-format off and reformat.

File: meta-facebook/yv3-vf/prj.conf (hunk 1 above)
Comment (HungYi-Li) [reply meta-facebook/yv3-vf/prj.conf]: remove it.

File: meta-facebook/yv3-vf/prj.conf [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# cmsis-os2
+CONFIG_CMSIS_RTOS_V2=y
+CONFIG_NUM_PREEMPT_PRIORITIES=56
+CONFIG_HEAP_MEM_POOL_SIZE=32768
+CONFIG_CMSIS_V2_MSGQ_MAX_DYNAMIC_SIZE=8192
+CONFIG_POLL=y
+CONFIG_THREAD_NAME=y
+CONFIG_THREAD_STACK_INFO=y
+CONFIG_THREAD_MONITOR=y
+CONFIG_IRQ_OFFLOAD=y
```
[the remark this answers is not in the document]
Comment (HungYi-Li) [reply meta-facebook/yv3-vf/prj.conf]: remove it.

**Comment by HungYi-Li** [comment]
Hi @GoldenBug ,
I modified it based on the comments.
Please review again, thank you

**Pull request closed without merging**